lgc/builder/DescBuilder.cpp

```diff
@@ -239,10 +279,21 @@ Value *DescBuilder::CreateGetDescPtr(ResourceNodeType concreteType, ResourceNode
   if (!m_pipelineState->isUnlinked() || !m_pipelineState->getUserDataNodes().empty()) {
     std::tie(topNode, node) = m_pipelineState->findResourceNode(abstractType, descSet, binding);
     if (!node && m_pipelineState->findResourceNode(ResourceNodeType::Unknown, descSet, binding).second) {
-      // NOTE: Resource node may be DescriptorTexelBuffer, but it is defined as OpTypeSampledImage in SPIRV,
-      // In this case, a caller may search for the DescriptorSampler and not find it. We return nullptr and
-      // expect the caller to handle it.
-      return UndefValue::get(getDescPtrTy(concreteType));
+      if (!node) {
+        // If we can't find the node, assume mutable descriptor and search for any node.
+        std::tie(topNode, node) = m_pipelineState->findResourceNode(ResourceNodeType::Unknown, descSet, binding);
+        if (!node) {
+          // We did not find the resource node. Return an undef value.
+          return UndefValue::get(getDescPtrTy(concreteType));
+        }
+        if (!node && m_pipelineState->findResourceNode(ResourceNodeType::Unknown, descSet, binding).second) {
+          // NOTE: Resource node may be DescriptorTexelBuffer, but it is defined as OpTypeSampledImage in SPIRV,
```

Comment:
Same here, this can never happen.

Reply:
Removed, we now check for the DescriptorMutable type